shell/platform/windows/task_runner.h

```diff
@@ -5,35 +5,61 @@
 #ifndef FLUTTER_SHELL_PLATFORM_WINDOWS_TASK_RUNNER_H_
 #define FLUTTER_SHELL_PLATFORM_WINDOWS_TASK_RUNNER_H_
 
-#include <windows.h>
-
 #include <chrono>
+#include <deque>
+#include <functional>
 #include <memory>
+#include <mutex>
+#include <queue>
+#include <variant>
 
 #include "flutter/shell/platform/embedder/embedder.h"
 
 namespace flutter {
 
 typedef uint64_t (*CurrentTimeProc)();
 
-// Abstract custom task runner for scheduling custom tasks.
-class TaskRunner {
+// Abstract custom timer.
+class TaskRunnerTimer {
+ public:
+  class Delegate {
+   public:
+    virtual std::chrono::nanoseconds ProcessTasks() = 0;
```

Comment:
Please add some docs for this method and the other interface methods below. Also any reason for creating this base class rather than have these virtual methods exist on `TaskRunner` itself? As it stands right now, `TaskRunnerTimer` (which reads like a timer), has methods which aren't typically on timers, e.g. `WakeUp` and `RunsOnCurrentThread`. If these methods were to be on `TaskRunner` base class itself, and have `TaskRunnerWin32` and `TaskRunnerWinUwp` implement this, it would make more sense IMO.

Reply:
I'll add the docs soon.
I wanted to clarify the callback in `TashRunnerTimer`, but I also agree that it seems not like timer.